docs/sources/plugins/developing/auth-for-datasources.md

```diff
@@ -9,89 +9,102 @@ weight = 3
 
-## Plugin Routes
+## Encrypting sensitive data
+
+When a user saves a password or secret with your data source plugin's Config page, then you can save data in an encrypted blob in the Grafana database called `secureJsonData`. Any data saved in the blob is encrypted by Grafana and can only be decrypted by the Grafana server on the backend. This means once a password is saved, no sensitive data is sent to the browser. If the password is saved in the `jsonData` blob or the `password` field then it is unencrypted and anyone with Admin access (with the help of Chrome Developer Tools) can read it.
 
-You can specify routes in the `plugin.json` file for your data source plugin. [Here is an example](https://github.com/grafana/azure-monitor-datasource/blob/d74c82145c0a4af07a7e96cc8dde231bfd449bd9/src/plugin.json#L30-L95) with lots of routes (though most plugins will just have one route).
+This is an example of using the `secureJsonData` blob to save a property called `password` in a html input:
+
+```html
+<input type="password" class="gf-form-input" ng-model="ctrl.current.secureJsonData.password" placeholder="password" />
+```
+
+## Plugin routes
+
+A plugin route describes where the intercepted request should be forwarded to and how to authenticate for the external API. You can define multiple routes that can match multiple external API endpoints.
+
+You specify routes in the `plugin.json` file for your data source plugin. [Here is an example](https://github.com/grafana/azure-monitor-datasource/blob/d74c82145c0a4af07a7e96cc8dde231bfd449bd9/src/plugin.json#L30-L95) with lots of routes (though most plugins will just have one route).
 
 When you build your URL to the third-party API in your data source class, the URL should start with the text specified in the path field for a route. The proxy will strip out the path text and replace it with the value in the URL field.
 
-For example, if my code makes a call to URL `azuremonitor/foo/bar` with this code:
+### Simple plugin route example
 
-```js
-this.backendSrv.datasourceRequest({
-  url: url,
-  method: 'GET',
-})
-```
+- If my code makes a call to URL `azuremonitor/foo/bar` with this code:
 
-and this route:
+  ```js
+  this.backendSrv.datasourceRequest({
+    url: url,
+    method: "GET",
+  });
+  ```
 
-```json
-"routes": [{
-  "path": "azuremonitor",
-  "method": "GET",
-  "url": "https://management.azure.com",
-  ...
-}]
-```
+- and the plugin has this route:
+
+  ```json
+  "routes": [{
+    "path": "azuremonitor",
+    "method": "GET",
+    "url": "https://management.azure.com"
+  }]
+  ```
 
-then the Grafana proxy will transform it into "https://management.azure.com/foo/bar" and add CORS headers.
+- then the Grafana proxy will transform the url from the original request into `https://management.azure.com/foo/bar`
+- finally, it will add CORS headers and forward the request to the new url. This example does not do any authentication.
```

Comment:
```suggestion
- then the Grafana proxy will transform the URL from the original request into `https://management.azure.com/foo/bar`
- finally, it will add CORS headers and forward the request to the new URL. This example does not do any authentication.
```